core/transformations.h

```diff
@@ -1914,6 +1914,78 @@ void radialAverageAxis(const MultidimArray< T >& in, char axis, MultidimArray< d
 		REPORT_ERROR(ERR_ARG_INCORRECT,"Not implemented yet");
 }
 
+template<typename T>
+void radialAverageNonCubic(const MultidimArray< T >& m,
+                   Matrix1D< int >& center_of_rot,
+                   MultidimArray< T >& radial_mean,
+                   MultidimArray< int >& radial_count,
+                   bool rounding = false)
+{
+    Matrix1D< double > idx(3);
+
+    size_t sizemax = std::max({XSIZE(m), YSIZE(m), ZSIZE(m)});
+    double scalex = XSIZE(m)/sizemax;
+    double scaley = YSIZE(m)/sizemax;
+    double scalez = ZSIZE(m)/sizemax;
+
+    // If center_of_rot was written for 2D image
+    if (center_of_rot.size() < 3)
+        center_of_rot.resize(3);
+
+    // First determine the maximum distance that one should expect, to set the
+    // dimension of the radial average vector
+    MultidimArray< int > distances(8);
+
+	const double z0 = STARTINGZ(m) - ZZ(center_of_rot);
+	const double y0 = STARTINGY(m) - YY(center_of_rot);
+	const double x0 = STARTINGX(m) - XX(center_of_rot);
+
+	const double xf = FINISHINGX(m) - XX(center_of_rot);
+	const double yf = FINISHINGY(m) - YY(center_of_rot);
+	const double zf = FINISHINGZ(m) - ZZ(center_of_rot);
+
+	distances(0) = (int) floor(sqrt(x0 * x0 + y0 * y0 + z0 * z0));
+	distances(1) = (int) floor(sqrt(xf * xf + y0 * y0 + z0 * z0));
+	distances(2) = (int) floor(sqrt(xf * xf + yf * yf + z0 * z0));
+	distances(3) = (int) floor(sqrt(x0 * x0 + yf * yf + z0 * z0));
+	distances(4) = (int) floor(sqrt(x0 * x0 + yf * yf + zf * zf));
+	distances(5) = (int) floor(sqrt(xf * xf + yf * yf + zf * zf));
+	distances(6) = (int) floor(sqrt(xf * xf + y0 * y0 + zf * zf));
+	distances(7) = (int) floor(sqrt(x0 * x0 + y0 * y0 + zf * zf));
+
+    int dim = CEIL(distances.computeMax());
+    if (rounding)
+        dim++;
```

Comment:
I'm still not sure if you actually need to do the `dim++`. What is the effect of `rounding==true`?
Because on line 1974 you do either `round` or `floor`, so if you do only `ceil` here, you should be covered, no?